types/jquery/index.d.ts

```diff
@@ -2389,12 +2390,24 @@ interface JQuery<TElement extends Node = HTMLElement> {
      * @since 1.4
      */
     wrapInner(wrappingElement: JQuery.Selector | JQuery.htmlString | Element | JQuery | ((this: TElement, index: number) => string | JQuery | Element)): this;
+
+    [n: number]: TElement;
 }
 
-interface JQuery<TElement extends Node = HTMLElement> extends ArrayLike<TElement>, Iterable<TElement> { }
+interface JQuery<TElement extends Node = HTMLElement> extends Iterable<TElement> { }
```

Comment:
Yes, but `ArrayLike`'s definition has the index signature marked as `readonly`.
```typescript
interface ArrayLike<T> {
    readonly length: number;
    readonly [n: number]: T;
}
```
I added an index signature to `JQuery` to compensate (it already has a length property).

It looks like writing through the index signature is a [real scenario](https://github.com/jquery/sizzle/wiki#sizzleselectorspseudosname--function-elem--).
```javascript
var $test = jQuery( document );
Sizzle.selectors.pseudos.fixed = function( elem ) {
    $test[ 0 ] = elem;
    return $test.css( "position" ) === "fixed";
};
```